v4/newrelic/transaction.go

```diff
@@ -109,14 +110,39 @@ func (txn *Transaction) AddAttribute(key string, value interface{}) {}
 // details on request attributes, url, and method.  If headers are
 // present, the agent will look for distributed tracing headers using
 // Transaction.AcceptDistributedTraceHeaders.
-func (txn *Transaction) SetWebRequestHTTP(r *http.Request) {}
+func (txn *Transaction) SetWebRequestHTTP(r *http.Request) {
+	if r == nil {
+		txn.SetWebRequest(WebRequest{})
+		return
+	}
+	wr := WebRequest{
+		Header:    r.Header,
+		URL:       r.URL,
+		Method:    r.Method,
+		Transport: transport(r),
+		Host:      r.Host,
+	}
+	txn.SetWebRequest(wr)
+}
 
 // SetWebRequest marks the transaction as a web transaction.  SetWebRequest
 // additionally collects details on request attributes, url, and method if
 // these fields are set.  If headers are present, the agent will look for
 // distributed tracing headers using Transaction.AcceptDistributedTraceHeaders.
 // Use Transaction.SetWebRequestHTTP if you have a *http.Request.
-func (txn *Transaction) SetWebRequest(r WebRequest) {}
+func (txn *Transaction) SetWebRequest(r WebRequest) {
+	txn.AcceptDistributedTraceHeaders(r.Transport, r.Header)
+}
+
+func transport(r *http.Request) TransportType {
+	if strings.HasPrefix(r.Proto, "HTTP") {
```

Comment:
I thought `Proto` can anyway only hold either `HTTP/1.1` or `HTTP/2`. But it doesn't harm to be cautious.

Reply:
Yeah, I'm completely unsure of this. I just copied this method from the `v3` agent, to keep things consistent when people upgrade. Plus, at this point, we don't even use the `Transport` argument sent to `AcceptDistributedTraceHeaders`. I debated leaving this out entirely for that reason, but decided to keep it around to make things easier on our future selves.